crypto/cpuid.c

```diff
@@ -125,24 +125,29 @@ void OPENSSL_cpuid_setup(void)
         } else if (env[0] == ':') {
             vec = OPENSSL_ia32_cpuid(OPENSSL_ia32cap_P);
         }
-
-        if ((env = ossl_strchr(env, ':')) != NULL) {
-            IA32CAP vecx;
-
-            env++;
-            off = (env[0] == '~') ? 1 : 0;
-            vecx = ossl_strtouint64(env + off);
-            if (off) {
-                OPENSSL_ia32cap_P[2] &= ~(unsigned int)vecx;
-                OPENSSL_ia32cap_P[3] &= ~(unsigned int)(vecx >> 32);
-            } else {
-                OPENSSL_ia32cap_P[2] = (unsigned int)vecx;
-                OPENSSL_ia32cap_P[3] = (unsigned int)(vecx >> 32);
+        int index = 2;
+        for (; index < OPENSSL_IA32CAP_P_MAX_INDEXES; index += 2) {
+            if (env && (((env = ossl_strchr(env, ':')) != NULL))) {
```

Comment:
An explicit `NULL` test for `env` (similar to the one 2 lines below) can increase the readability.

Reply:
Added a check